### ISSUE TITLE
Model generates aliphatic molecules despite aromatic training data

### ISSUE THREAD

**Issue by raragreen** [kxz18/UniMoMo#12]
We fine-tuned UniMoMo from scratch on 151 aromatic heterocycle molecules.
Despite h_loss_weight=20 and 8000+ epochs, generated molecules are 
almost entirely aliphatic (mean 0.4 aromatic atoms vs 12+ in training).

Is the pretrained autoencoder limiting aromatic generation?
How can we generate molecules matching our training data chemistry?

**Comment by kxz18** [comment]
Could you try to use the target in the training set for generation, to see whether it is generation aromatic molecules? From my perspective, a training set of 151 molecules for 8000+ epochs might be way too long for a generalizable convergence. It might already be very overfitted on your training complexes, and output random results on other targets that are not in the training set. Also, a h_loss_weight of 20 is far beyond the common scope. It should be similar to the x_loss, otherwise the generated latent coordinates might introduce severe distribution shift and make the VAE fail to work in a correct way.

**Comment by raragreen** [comment]
Thank you for your response! May I ask do I need to first train VAE before LDM training? Do you have a general good loss value for both VAE and LDM? I always get 0.6~1.0.

**Comment by raragreen** [comment]
For a training set of 151 molecules, Do you prefer train from scratch or the pre-trained model?

**Comment by kxz18** [comment]
For the loss scales, it really depends on the training distribution and your setting of $\lambda_1$ and $\lambda_2$. We have provided loss curves for LDM using VAEs with different $\lambda$ settings in Appendix C.2:

<img width="2304" height="1340" alt="Image" src="https://github.com/user-attachments/assets/e7f2919d-db29-44f4-b67b-541a43fbd27a" />

For the default setting, I think loss scales of 0.6~1.0 are ok.

As for the 151 molecules dataset, I would choose finetuning the pretrained model. However, I would like to point out this might be far beyond the minimum size of dataset that could support deep-learning models. Normally,  at least thousands of data points should be used for training a deep-learning model.
